docs(releases): add Stellar ITS v1.2.0 release doc (#741)
axelarnetwork/axelar-contract-deployments: releases/stellar/2025-05-ITS-v1.2.0.md

# Stellar ITS v1.2.0

|                | **Owner**                            |
| -------------- | ------------------------------------ |
| **Created By** | @ahram (<ahram@interoplabs.io>)   |
| **Deployment** | @ahram (<ahram@interoplabs.io>)   |

| **Network**          | **Deployment Status** | **Date**   |
| -------------------- | --------------------- | ---------- |
| **Devnet Amplifier** | TBD                   | 2025-05-12 |
| **Stagenet**         | TBD                   | 2025-05-12 |
| **Testnet**          | TBD                   | 2025-05-12 |
| **Mainnet**          | TBD                   | TBD        |

> - [Crates](https://crates.io/crates/stellar-interchain-token-service/1.2.0)

Reviewer: let's cut a release?

Author: Done: https://github.com/axelarnetwork/axelar-amplifier-stellar/actions/runs/14979447785/job/42079819013

- [Releases](https://github.com/axelarnetwork/axelar-amplifier-stellar/releases/tag/stellar-interchain-token-service-v1.2.0)

## Background

- This is the v1.2.0 Stellar ITS release.

Contract changes in the release:

> - emit data_hash instead of data in InterchainTransferSentEvent ([#342](https://github.com/axelarnetwork/axelar-amplifier-stellar/pull/342))
> - This is a minor bump since it breaks the event API for consumers.

Reviewer: ```suggestion
- ([#342](https://github.com/axelarnetwork/axelar-amplifier-stellar/pull/342)): Breaking change to the `InterchainTransferSentEvent` event in ITS. `data_hash` is emitted now instead of `data`.
```
describe the issue as breaking instead of making a separate note about it

```

1. Verify deployer address

| Network              | `deployer address`                                         |
| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | `GCRN3JXRVXHQTFQFM7NR4TTTORGZDCJWPIOLPQQHL6WMAQGVMWSXJL3Q` |
| **Stagenet**         | `GBP4FSAOFV5O72AB3YQRDCYVD47W4N7KQK3OJODXSU3OBPNGKX4SQTJ3` |
| **Testnet**          | `GBP4FSAOFV5O72AB3YQRDCYVD47W4N7KQK3OJODXSU3OBPNGKX4SQTJ3` |
| **Mainnet**          | `GC2SJ4YXCMP2LYXMXBNJMK6SNK4XUR7TGJXY4GA3VACNMCZVCQ6VFGG3` |

2. Upgrade `InterchainTokenService`

```bash
node stellar/deploy-contract.js upgrade InterchainTokenService --version 1.2.0
```

## Checklist

The following checks should be performed after the rollout

### Stellar → EVM

1. Deploy Native Interchain Token

```bash
node stellar/its.js deploy-interchain-token TEST2 test2 18 0x5678 100

node stellar/its.js deploy-remote-interchain-token 0x1234 [destination-chain] --gas-amount 10000000
```

2. Interchain Token Transfer for Native Interchain Token

```bash
node stellar/its.js interchain-transfer [token-id] [destination-chain] [destination-address] [amount] --gas-amount 10000000
```

> ### EVM → Stellar

Reviewer: this is not really important since the breaking change is for transfers from Stellar -> X

Reviewer: Specialize the checklist to the release being made instead of copy pasting. E.g. Ensure that the event being changed was emitted correctly.

Author: Make sense, added instructions how to check the vent and ensure the event being emitted correctly.
